libsql/src/value.rs

```diff
@@ -458,13 +458,76 @@ mod serde_ {
                     Ok(Value::Integer(v))
                 }
 
+                fn visit_i32<E>(self, v: i32) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    Ok(Value::Integer(v as i64))
+                }
+
+                fn visit_i16<E>(self, v: i16) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    Ok(Value::Integer(v as i64))
+                }
+
+                fn visit_i8<E>(self, v: i8) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    Ok(Value::Integer(v as i64))
+                }
+
+                fn visit_u64<E>(self, v: u64) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    if v > i64::MAX as u64 {
+                        Err(serde::de::Error::invalid_value(
+                            de::Unexpected::Unsigned(v),
+                            &"u64 is too large to fit in an i64",
+                        ))
+                    } else {
+                        Ok(Value::Integer(v as i64))
+                    }
+                }
+
+                fn visit_u32<E>(self, v: u32) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    Ok(Value::Integer(v as i64))
+                }
+
+                fn visit_u16<E>(self, v: u16) -> std::result::Result<Self::Value, E>
+                where
+                    E: serde::de::Error,
+                {
+                    Ok(Value::Integer(v as i64))
+                }
```

Comment:
shouldn't those already be handled by the delegation?

Reply:
I don't believe so, i got an error when deserializing an integer other than i64 which was solved by this